packages/extension/src/serviceWorker/index.ts

```diff
@@ -17,790 +17,373 @@
- * @todo Shouldn't have to reinstall the extension.
- */
-chrome.runtime.onInstalled.addListener(async (details) => {
-  syncCookieStore = new SynchnorousCookieStore();
-  syncCookieStore?.clear();
-
-  setInterval(async () => {
-    await chrome.storage.session.get();
-  }, 20000);
-
-  if (details.reason === 'install') {
-    await chrome.storage.sync.clear();
-    await chrome.storage.sync.set({
-      allowedNumberOfTabs: 'single',
-      isUsingCDP: false,
-    });
-  }
-
-  if (details.reason === 'update') {
-    await chrome.storage.local.clear();
-    const preSetSettings = await chrome.storage.sync.get();
-    tabMode = preSetSettings?.allowedNumberOfTabs ?? 'single';
-    globalIsUsingCDP = preSetSettings?.isUsingCDP ?? false;
-    if (
-      preSetSettings?.allowedNumberOfTabs &&
-      Object.keys(preSetSettings).includes('isUsingCDP')
-    ) {
-      return;
-    }
-
-    await chrome.storage.sync.clear();
-    await chrome.storage.sync.set({
-      allowedNumberOfTabs: 'single',
-      isUsingCDP: false,
-    });
-  }
-});
-
-/**
- * Fires whenever debugging target issues instrumentation event.
- * @see https://developer.chrome.com/docs/extensions/reference/api/debugger
- */
-// eslint-disable-next-line complexity
-chrome.debugger.onEvent.addListener((source, method, params) => {
-  if (!ALLOWED_EVENTS.includes(method) || !source?.tabId || !params) {
-    return;
-  }
-
-  const url = syncCookieStore?.getTabUrl(source?.tabId);
-
-  const tabId = source.tabId.toString();
-
-  if (tabMode !== 'unlimited' && tabToRead !== tabId) {
-    return;
-  }
-
-  if (method === 'Network.responseReceived') {
-    const request = params as Protocol.Network.ResponseReceivedEvent;
-
-    // To get domain from the request URL if not given in the cookie line.
-    if (!requestIdToCDPURLMapping[tabId]) {
-      requestIdToCDPURLMapping[tabId] = {
-        [request.requestId]: request?.response.url,
-      };
-    } else {
-      requestIdToCDPURLMapping[tabId] = {
-        ...requestIdToCDPURLMapping[tabId],
-        [request.requestId]: request?.response.url,
-      };
-    }
-  }
+      await Promise.all(
+        targets.map(async ({ id, url }) => {
+          if (url.startsWith('http')) {
+            await attachCDP({ targetId: id });
+          }
+        })
+      );
 
-  if (method === 'Network.requestWillBeSentExtraInfo') {
-    const requestParams =
-      params as Protocol.Network.RequestWillBeSentExtraInfoEvent;
+      // This is to get a list of all targets being attached to the main frame.
+      if (method === 'Target.attachedToTarget' && params) {
+        const {
+          targetInfo: { targetId, url },
+        } = params as Protocol.Target.AttachedToTargetEvent;
 
-    const { associatedCookies, requestId } = requestParams;
+        await attachCDP({ targetId });
 
-    if (associatedCookies.length === 0) {
-      return;
-    }
+        targets = await chrome.debugger.getTargets();
```

Comment:
Will it return a different result than what already being called on line 61?

Reply:
Yes. In between this and the line 61 if there is a frame added to the list there is a chance that we might miss this. Although the chances are less but this is done to be on a safer side.